src/pubkeyaction.cpp

```diff
@@ -108,16 +109,17 @@ void PubKeyActionCreateShare::proc(MegaClient* client, User* u)
 
     // we have all ingredients ready: the target user's public key, the share
     // key and all nodes to share
-    client->restag = tag;
-    client->reqs.add(new CommandSetShare(client, n, u, a, newshare, NULL, selfemail.c_str()));
+    client->reqs.add(new CommandSetShare(client, n, u, a, newshare, NULL, selfemail.c_str(), tag, move(completion)));
 }
 
 // share node sh with access level sa
-PubKeyActionCreateShare::PubKeyActionCreateShare(handle sh, accesslevel_t sa, int ctag, const char* personal_representation)
+PubKeyActionCreateShare::PubKeyActionCreateShare(handle sh, accesslevel_t sa, int ctag, const char* personal_representation, std::function<void(Error)> f)
```

Comment:
Now that `personal_representation` won't be nullptr by default, we may want to adjust the test:
`if (personal_representation)` to become `if (personal_representation && personal_representation[0])`.

Reply:
My idea about that was just that anywhere we were using the default before, we would pass nullptr now, so it's all the same in that respect.   Scanning through now, I think it's only used in setshare() anyway, and that parameter is passed from setshare()'s parameter of the same name, and so in fact the default was never used anyway.